core/iwasm/libraries/libc-wasi/libc_wasi_wrapper.c

```diff
@@ -969,9 +1007,48 @@ wasi_poll_oneoff(wasm_exec_env_t exec_env, const wasi_subscription_t *in,
         || !validate_native_addr(nevents_app, sizeof(uint32)))
         return (wasi_errno_t)-1;
 
-    err = wasmtime_ssp_poll_oneoff(curfds, in, out, nsubscriptions, &nevents);
-    if (err)
-        return err;
+    __wasi_timestamp_t elapsed = 0;
+    const __wasi_timestamp_t timeout = get_timeout_for_poll_oneoff(
+                                 in, nsubscriptions),
+                             time_quant = 1e9;
+    const uint64 size_to_copy = nsubscriptions * sizeof(wasi_subscription_t);
```

Comment:
Had better use `nsubscriptions * (uint64)sizeof(wasi_subscription_t)` to avoid possible integer overflow.

Reply:
done